pkg/extension/sumologicextension/extension.go

```diff
@@ -268,6 +281,63 @@ func (pm *sumologicExtension) register(ctx context.Context, collectorName string
 	return nil
 }
 
+func (pm *sumologicExtension) heartbeat(ctx context.Context) {
+	heartbeatInterval := defaultHeartbeatInterval
+	if pm.conf.HeartBeatInterval > 0 {
+		heartbeatInterval = pm.conf.HeartBeatInterval
+	}
```

Comment:
I'd do this in `newSumologicExtension` or in `createDefaultConfig` in `factory.go` with preference for the latter.